cognite/seismic/protos/v1/seismic_service_datatypes.proto

```diff
@@ -300,3 +300,38 @@ enum TraceHeaderField {
     CROSSLINE = 4;
     SHOTPOINT = 5;
 }
+
+message GeometryFilter {
+    Geometry geometry = 1;
+    InterpolationMethod interpolation_method = 2; // Only used for linear geometries
+}
+
+message Seismic2dExtent {
+    TraceHeaderField trace_key = 1; // Must be an applicable field for 2D
+    LineDescriptor trace_range = 2; // Range of header values to use
+}
+
+message Seismic3dExtent {
+    oneof type {
+        Seismic3dRects rects = 1;
+        Seismic3dDef def = 2;
+    }
+}
+
+message Seismic3dRects {
+    repeated Seismic3dRect rects = 1;
+}
+
+message Seismic3dRect {
+    LineDescriptor inline_range = 1;
+    LineDescriptor xline_range = 2;
+}
+
+message Seismic3dDef {
+    TraceHeaderField major_direction = 1; // Must be inline or crossline
```

Comment:
It feels a bit odd to me that the single `TraceHeaderField` for the 3d def to be considered a direction when it's just describing the Major. Having both major and minor fields and only accepting Inline/Crossline Crossline/Inline combinations seems okay to me, and would support other combinations in the future (I have no idea if this would ever be a requirement, maybe not?)

Reply:
Yeah, maybe it's fine.